### PR TITLE
Kernel: Update kernel config

### PR DESCRIPTION
**Pull request by jongwu**
Lots of kernel configs removed in latest kernel. Remove them in kata kernel config.

Remove list:
CONFIG_SECURITY_SELINUX_CHECKREQPROT_VALUE
CONFIG_IP_NF_TARGET_CLUSTERIP
CONFIG_NET_SCH_CBQ
CONFIG_RANDOM_TRUST_CPU

As CONFIG_RANDOM_TRUST_CPU is removed, we need pass corresponding kernel parameter to dragonball on aarch64 to use CPU's rng.

Fixes: #8408

### CODE REVIEW

**Review by cmaf — APPROVED** [review]
LGTM @jongwu

**Comment by jongwu** [comment]
@studychao -, TAL?

**Review by studychao — APPROVED** [review]
Thanks, @jongwu LGTM!
Out of curiosity, do cloud hypervisor / QEMU need this kernel params too to enable rng?

**Comment by jongwu** [comment]
> Thanks, @jongwu LGTM! Out of curiosity, do cloud hypervisor / QEMU need this kernel params too to enable rng?

No. There is no RNG device inside DB but CH/QEMU has.

**Comment by jongwu** [comment]
> > Thanks, @jongwu LGTM! Out of curiosity, do cloud hypervisor / QEMU need this kernel params too to enable rng?
> 
> No. There is no RNG device inside DB but CH/QEMU has.

Is there any plan to add virtio-rng inside DB?

**Comment by studychao** [comment]
> > > Thanks, @jongwu LGTM! Out of curiosity, do cloud hypervisor / QEMU need this kernel params too to enable rng?
> > 
> > 
> > No. There is no RNG device inside DB but CH/QEMU has.
> 
> Is there any plan to add virtio-rng inside DB?

We got plan but not in the highest priority :)

**Comment by jongwu** [comment]
It slows down the guest boot if there is no RNG dev on aarch64. I tried to add an virtio-rng in DB last year but doesn't push it to DB community. Luckily, I find other way to mitigate it.

**Comment by studychao** [comment]
> It slows down the guest boot if there is no RNG dev on aarch64. I tried to add an virtio-rng in DB last year but doesn't push it to DB community. Luckily, I find other way to mitigate it.

ohhh, if you already have an implementation of virtio-rng, could you please push that to the community?

**Comment by jongwu** [comment]
I have not got the legal apprval to do that inside Arm which is part of Arm policy. That's the reason.

**Comment by studychao** [comment]
> I have not got the legal apprval to do that inside Arm which is part of Arm policy. That's the reason.

got that~

**Comment by studychao** [comment]
LGTM in this path!

**Comment by jongwu** [comment]
Thanks! @studychao

File: tools/packaging/kernel/configs/fragments/arm64/base.conf [hunk 1]
```diff
@@ -43,7 +43,6 @@ CONFIG_ARM64_RAS_EXTN=y
 # ARMv8.5 architectural features
 #
 CONFIG_ARCH_RANDOM=y
-CONFIG_RANDOM_TRUST_CPU=y
```
Comment (amshinde) [inline tools/packaging/kernel/configs/fragments/arm64/base.conf:46]: Since we had this enabled, should the kernel param be passed for all hypervisors?

Comment (jongwu) [reply tools/packaging/kernel/configs/fragments/arm64/base.conf:46]: Err, IMO, CPU RNG is a substitution for RNG device. It seems only DB lack of RNG device, I think we can only pass the parameter for DB.

Comment (amshinde) [reply tools/packaging/kernel/configs/fragments/arm64/base.conf:46]: ok, thanks for clarifying.

**Comment by amshinde** [comment]
@jongwu Can you rebase this PR, so that we can get it merged?

**Comment by jongwu** [comment]
> @jongwu Can you rebase this PR, so that we can get it merged?

Thanks @amshinde -, Done.

**Pull request closed without merging**